cli_client/python/timesketch_cli_client/commands/search.py

```diff
@@ -0,0 +1,206 @@
+
+    Args:
+        search_obj: API Search object.
+    """
+    click.echo('Query string: {}'.format(search_obj.query_string))
+    click.echo('Return fields: {}'.format(search_obj.return_fields))
+    click.echo('Filter: {}'.format(
+        json.dumps(search_obj.query_filter, indent=2)))
+
+
+@click.command('search')
+@click.option(
+    '--query', '-q', default='*',
+    help='Search query in Elasticsearch query string format')
+@click.option(
+    '--time', 'times', multiple=True,
+    help='Datetime filter (e.g. 2020-01-01T12:00)')
+@click.option(
+    '--time-range', 'time_ranges', multiple=True, nargs=2,
+    help='Datetime range filter (e.g: 2020-01-01 2020-02-01)')
+@click.option(
+    '--label', 'labels', multiple=True,
+    help='Filter events with label')
+@click.option(
+    '--header/--no-header', default=True,
+    help='Toggle header information (default is to show)')
+@click.option(
+    '--output-format', 'output',
+    help='Set output format (overrides global setting)')
+@click.option(
+    '--return-fields', 'return_fields', default='',
+    help='What event fields to show')
+@click.option(
+    '--order', default='asc',
+    help='Order the output (asc/desc) based on the time field')
+@click.option(
+    '--limit', type=int, default=40,
+    help='Limit amount of events to show (default: 40)')
+@click.option(
+    '--saved-search', type=int, help='Query and filter from saved search')
+@click.option(
+    '--describe', is_flag=True, default=False,
+    help='Show the query and filter then exit')
+@click.pass_context
+# pylint: disable=too-many-arguments
+def search_group(ctx, query, times, time_ranges, labels, header, output,
+                 return_fields, order, limit, saved_search, describe):
+    """Search and explore."""
+    sketch = ctx.obj.sketch
+    output_format = ctx.obj.output_format
+    search_obj = search.Search(sketch=sketch)
+
+    if output:
+        output_format = output
+
+    new_line = True
+    if output_format == 'csv':
+        new_line = False
+
+    # Construct query from saved search and return early.
+    if saved_search:
+        search_obj.from_saved(saved_search)
+        if describe:
+            describe_query(search_obj)
+            return
+        click.echo(format_output(
+            search_obj, output_format, header), nl=new_line)
+        return
+
+    # Construct the query from flags.
+    search_obj.query_string = query
+    search_obj.return_fields = return_fields
+    search_obj.max_entries = limit
+
+    if order == 'asc':
+        search_obj.order_ascending()
+    elif order == 'desc':
+        search_obj.order_descending()
+
+    # TODO: Add term chips.
+    if time_ranges:
+        for time_range in time_ranges:
+            range_chip = search.DateRangeChip()
+            range_chip.start_time = time_range[0]
+            range_chip.end_time = time_range[1]
+            search_obj.add_chip(range_chip)
+
+    if times:
+        for time in times:
```

Comment:
so for each time passed in you create a range chip with both the start/end time the exact time you are adding? is that what we want, just to limit to that exact second?

Why not use the DateIntervalChip and do a 1 minute or something before/after or have that as a configurable item as well?

Reply:
Tne time flag only accepts one, and its a single time. So this means that the user wants to filter on that. But... I see what you are getting at: If I'm not abusing the time chip with non-ISO dates then this will become a problem.

Let me add a TODO and followup on this,

Reply:
I would opt for either using the interval here, with then a default value or adding a parameter to set the range... since we want to make sure that the timestamp is ISO formatted... also you need to verify that here.

The reason is that if you save a search object with a chip that isn't ISO formatted, then `you'll be in trouble` in the API client, since the API client expects ISO formatted dates, otherwise chips will fail

Reply:
this has not been addressed

Reply:
Fixed for now, and will revisit later.